api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -575,3 +620,33 @@ func checkCatalog(catalog *model.CatalogState) bool {
 	}
 	return false
 }
+
+func (i *MaterializeStageProvider) filterIncompleteDeployment(ctx context.Context, namespace string, objectNames []string, isInstance bool) []string {
```

Comment:
I wonder if this method should be extended so that if we found there are unrecoverable errors we should stop waiting

Reply:
You are right. I will check if deployment reached termination state and then check if deployment failed